include/zephyr/drivers/i3c.h

```diff
@@ -0,0 +1,2301 @@
+ *
+ * Dumps out a list of I3C messages. For any that are writes (W), the data is
+ * displayed in hex.
+ *
+ * It looks something like this (with name "testing"):
+ *
+ * D: I3C msg: testing, addr=56
+ * D:    W len=01:
+ * D: contents:
+ * D: 06                      |.
+ * D:    W len=0e:
+ * D: contents:
+ * D: 00 01 02 03 04 05 06 07 |........
+ * D: 08 09 0a 0b 0c 0d       |......
+ *
+ * @param name Name of this dump, displayed at the top.
+ * @param msgs Array of messages to dump.
+ * @param num_msgs Number of messages to dump.
+ * @param target I3C target device descriptor.
+ */
+void i3c_dump_msgs(const char *name, const struct i3c_msg *msgs,
+		   uint8_t num_msgs, struct i3c_device_desc *target);
+
+/**
+ * @brief Dump out an I2C message
+ *
+ * Dumps out a list of I2C messages. For any that are writes (W), the data is
+ * displayed in hex.
+ *
+ * It looks something like this (with name "testing"):
+ *
+ * D: I2C msg: testing, addr=56
+ * D:    W len=01:
+ * D: contents:
+ * D: 06                      |.
+ * D:    W len=0e:
+ * D: contents:
+ * D: 00 01 02 03 04 05 06 07 |........
+ * D: 08 09 0a 0b 0c 0d       |......
+ *
+ * @param name Name of this dump, displayed at the top.
+ * @param msgs Array of messages to dump.
+ * @param num_msgs Number of messages to dump.
+ * @param i2c_dev I2C device descriptor.
+ */
+static inline void i3c_i2c_dump_msgs(const char *name,
+				     const struct i2c_msg *msgs,
+				     uint8_t num_msgs,
+				     struct i3c_i2c_device_desc *i2c_dev)
+{
+	i2c_dump_msgs(name, msgs, num_msgs, i2c_dev->addr);
+}
+
+/**
+ * @brief Generic helper function to perform bus initialization.
+ *
+ * @param dev Pointer to controller device driver instance.
+ * @param i3c_dev_list Pointer to I3C device list.
+ *
+ * @retval 0 If successful.
+ * @retval -EBUSY Bus is busy.
+ * @retval -EIO General input / output error.
+ * @retval -ENODEV If a provisioned ID does not match to any target devices
+ *                 in the registered device list.
+ * @retval -ENOSPC No more free addresses can be assigned to target.
+ * @retval -ENOSYS Dynamic address assignment is not supported by
+ *                 the controller driver.
+ */
+int i3c_bus_init(const struct device *dev, struct i3c_dev_list *i3c_dev_list);
+
+/**
+ * @brief Get basic information from device and update device descriptor.
+ *
+ * This retrieves some basic information:
+ *   * Bus Characteristics Register (GETBCR)
+ *   * Device Characteristics Register (GETDCR)
+ *   * Max Read Length (GETMRL)
+ *   * Max Write Length (GETMWL)
+ * from the device and update the corresponding fields of the device
+ * descriptor.
+ *
+ * This only updates the field(s) in device descriptor
+ * only if CCC operations succeed.
+ *
+ * @param[in,out] target I3C target device descriptor.
+ *
+ * @retval 0 if successful.
+ * @retval -EIO General Input/Output error.
+ */
+int i3c_device_basic_info_get(struct i3c_device_desc *target);
+
+struct i3c_target_callbacks {
```

Comment:
It looks like this is all unused for now. is this section here for i3c_target to be implemented in the future?